tests/test_dimension.py

```diff
@@ -120,6 +120,29 @@ class TestConditionalDimension(object):
     """A collection of tests to check the correct functioning of
     :class:`ConditionalDimension`s."""
 
+    def test_no_index(self):
+        nt = 19
+        grid = Grid(shape=(11, 11))
+        time = grid.time_dim
+
+        u = TimeFunction(name='u', grid=grid)
+        assert(grid.stepping_dim in u.indices)
+
+        u2 = Function(name='energy', grid=grid)
+
+        factor = 4
+        time_subsampled = ConditionalDimension('t_sub', parent=time, factor=factor)
+
+        eqns = [Eq(u.forward, u + 1), Eq(u2, u2 + u*u*time_subsampled)]
```

Comment:
So this is actually going to scale the u^2 by (time/4), right? - which is probably not your intention, based on the names of the variables?  To prevent that would require constructing an expression which always evaluated to 1, but which sympy would not simplify away - that seems kind of shaky.

I think if we want to use ConditionalDimensions this way - and some people probably do - we might need to use something akin to region= to express that requirement?

Reply:
This is a dummy test just to check it works as it should. I actually use it in a different way for our SLIM stuff.

Reply:
>use something akin to region= to express that requirement?
There is more than a few conditional cases (straight if conditoons/ time-regions/...) that need to be figured out yeah, this is just a step to make the currently supported case more robust.